spec/features/supply_spec.rb

```diff
@@ -0,0 +1,106 @@
+require 'rails_helper'
+
+describe "supply", type: :feature, js: true do
+
+  let(:circle) { create(:circle) }
+
+  let!(:circle_admin_role)      { create :circle_role_admin, circle: circle }
+
+  let(:user_1) { create(:user, primary_circle: circle) } #Admin
+  let(:user_2) { create(:user, primary_circle: circle) } #Volunteer
+
+  let(:public_group)  { create(:working_group, is_private: false, circle: circle) }
+  let(:private_group) { create(:working_group, is_private: true , circle: circle) }
+
+  def create_supply(name, description, due_date, location)
+    date = I18n.l(due_date) if due_date != nil
+    fill_in "Name of supply",                   with: name
+    fill_in "Description",                      with: description
+    fill_in "Due Date",                         with: date                      
+    fill_in "Location",                         with: location
+    
+    click_button  "Create Supply"
+  end
+
+  def visit_add_supply(user)
+    visit(circle_path(circle, as: user))
+    find('.button-super', text: /Add/).click
+    click_on('Supply')
+  end
+
+  before do
+    create(:circle_role_admin, circle: circle, user: user_1) 
+    create(:circle_role_volunteer, circle: circle, user: user_2)
+
+    public_group.roles.member.create              user: user_1
+    public_group.roles.member.create              user: user_2
+
+    private_group.roles.member.create             user: user_1
+  end
+
+  context 'add supply button' do
+    it 'visible to Admin' do
+      visit(circle_path(circle, as: user_1))
+      
+      expect(find('.button-super', text: /Add/)).to_not eq(nil)
+      find('.button-super', text: /Add/).click
+      expect(page).to have_link('Supply')
+    end
+
+    it 'not visible to Volunteer' do
+      visit(circle_path(circle, as: user_2))
+      expect(page).to_not have_selector('.button-super', text: /Add/)
+    end
+  end
+
+  context 'an admin' do
+
+    it "adds a supply" do
+      name        = "Test Supply"
```

Comment:
These could be removed since the above factory already sets some useful values. Override those or create a supply factory of your own (nested factory, https://goo.gl/xeQY7s) if you need to set specific values.